ament_cmake_google_benchmark/ament_cmake_google_benchmark/__init__.py

```diff
@@ -144,6 +139,63 @@ def convert_google_benchark_to_jenkins_benchmark(
     return out_data
 
 
+def convert_aggregate_benchmark(in_data):
+    out_data = {
+        'parameters': {
+            'repetitions': {
+                'value': in_data['repetitions'],
+            },
+        },
+    }
+
+    out_data.update(convert_extra_metrics(in_data, common_aggregate_test_properties))
+
+    return out_data
+
+
+def convert_iteration_benchmark(in_data):
+    out_data = {
+        'parameters': {
+            'iterations': {
+                'value': in_data['iterations'],
+            },
+            'repetitions': {
+                'value': in_data['repetitions'],
+            },
+        },
+        'cpu_time': {
+            'dblValue': in_data['cpu_time'],
+            'unit': in_data['time_unit'],
+        },
+        'real_time': {
+            'dblValue': in_data['real_time'],
+            'unit': in_data['time_unit'],
+        },
+    }
+
+    out_data.update(convert_extra_metrics(in_data, common_iteration_test_properties))
+
+    return out_data
+
+
+def convert_extra_metrics(in_data, common_properties):
```

Comment:
I got rather hung up on "common properties" being the name of the item set being skipped by this generator when the calling function is at least to my eyes, being used to add properties which are common to both iteration and aggregation benchmarks. So there was a mismatch between what I expected to find and what I thought I was looking at until I read more closely.

I haven't yet come up with anything both compact and precise, `handled_properties` or `properties_to_skip` both have potential.

Another way that might remove the issue altogether. Can you see a way for a single generator to build either benchmark type then we could just pop the whole thing in one and remove the split between type-specific and common properties handling.

Reply:
This code started as a generator, but it got so thick with the double-nested conditionals to determine the key name that I just gave up and made it an iterable function.

You're right about the variable name, though, I'll change that.